### PR TITLE
Added basic icon functionality

### PR DESCRIPTION
**Pull request by umbobabo**
The idea behind this implementation is to simplify implementation for buttons/links with icons too keeping more consistency across projects.

### CODE REVIEW

File: index.es6 [hunk 1]
```diff
@@ -14,11 +18,19 @@ export default function Button(props) {
...
+  const linkProps = {...props};
+  if (icon) {
+    extraClassNames.push('link-button--icon');
+    content = [ (<Icon {...icon} />), (<span className="link-button__text">{content}</span>) ];
+    // We don't want this prop spreaded on <a> tag.
+    delete linkProps.icon;
+  }
   return (
-    <a role="button" {...props} onClick={onClick}
+    <a role="button" {...linkProps} onClick={onClick}
```
Comment (danielmoll) [inline index.es6]: Accessibility wise there maybe a case where you'd rather have an actual `<button>` tag - ideally a button is used when the interaction doesn't take you away from the page - think hamburger, for example. Although you have the `role="button"` attribute I'm unsure about how that's handled in screen readers/voice over.

Comment (fabiosantoscode) [inline index.es6]: The idea behind using an `<a>` tag is to have a good non-javascript fallback. This component needs a `href` prop, always, for this reason.

File: package.json [hunk 2]
```diff
@@ -128,6 +128,7 @@
   "dependencies": {
     "@economist/component-devpack": "^3.5.0",
     "@economist/component-grid": "^1.3.0",
+    "@economist/component-icon": "^4.0.0",
```
Comment (fabiosantoscode) [inline package.json]: Are you using 4.0 or 5.0 now?

**Comment by fabiosantoscode** [comment]
@umbobabo If I could marry a pull request,
I wouldn't marry it anyway because real people are best.
But if there were a nuclear war and no one was left
I would
Refrigerator.

**Pull request merged**